README.md

```diff
@@ -1,15 +1,357 @@
+
+## Terminology
+
+There are several components related to Batcher...
+
+- __Batcher__: You will create one Batcher for each datastore that has capacity you wish to respect. Lots of Watcher will share the same Batcher. Batchers are long-lived.
+
+- __Operation__: You will enqueue Operations into Batcher. An Operation has an associated "Watcher", a "cost", a designation of whether or not it can be batched, a counter for the number of times it has been "attempted", and a "payload" (which can be anything you want).
+
+- __Watcher__: You will create one Watcher per process that you wish to manage. The Watcher receives the batches as they become available. Watchers are short-lived. For instance, if your solution is an HTTP server, you will probably create a Watcher with each request, send your Operations to a shared Batcher, and then get batches for processing back on your Watcher. If you need to handle different types of Operations that are processed in different ways or if they have different characteristics (such as an optimal batchsize), you might create a separate Watcher for each of those use-cases.
+
+There are also 2 rate limiters provided out-of-the-box...
+
+- __ProvisionedResource__: This is a simple rate limiter that has a fixed capacity per second.
+
+- __AzureSharedResource__: This rate limiter allows you to reserve a fixed amount of capacity and then share a fixed amount of capacity across multiple processes.
+
+Some other terms will be used throughout...
+
+- __Target__: As Operations are enqueued or marked done in Batcher it updates a Target number which is the total cost Batcher thinks is necessary to process any outstanding Operations. In other words, as Operations are enqueued, the Target grows by the cost of that Operation. When a batch is marked done, the Target is reduced by the cost of all Operations in that batch.
+
+- __Capacity__: The capacity that the rate limiter has been able to procure is available via the Capacity() method or the capacity event.
+
+- __MaxCapacity__: When using a rate limiter the MaxCapacity is the maximum capacity that could even be provided. For AzureSharedResource, this is the total of shared capacity and reserved capacity. For ProvisionedResource, this is the capacity that was provided when New() was called.
```

Comment:
The terms "shared capacity" and "reserved capacity" have not really been introduced by this point in the document and I don't think are going to be obvious to everyone.

Reply:
I have added those terms now.